src/components/TodoList/TodoList.tsx

```diff
@@ -1,208 +1,100 @@
 /* eslint-disable max-len */
 import React from 'react';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { actions as curentTodoFeat } from '../../features/currentTodo';
+import { Todo } from '../../types/Todo';
+import { FilterStatus } from '../../types/FilterStatus';
 
 export const TodoList: React.FC = () => {
+  const dispatch = useAppDispatch();
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+  const filter = useAppSelector(state => state.filter);
+  const todos = useAppSelector(state => state.todos.filter(
+    (todo: Todo) => {
+      return todo.title.toLocaleLowerCase().includes(filter.query.toLocaleLowerCase());
+    },
+  ));
+
+  const isSelected = (id: number) => currentTodo?.id === id;
+
+  const visibleTodos = () => {
```

Comment:
Use useMemo()  here. It ensures that the filtering process is performed only when necessary, reducing unnecessary calculations and improving your component's performance.